lib/chart_helpers.dart

```diff
@@ -93,3 +127,77 @@ ChartConfiguration generateBarChartConfig(
       data: dataset,
       options: _generateBarChartOptions(dataNormalisationEnabled));
 }
+
+ChartOptions _generateTimeSeriesChartOptions(bool dataNormalisationEnabled) {
+  var labelPrepend = dataNormalisationEnabled ? 'Percentage' : 'Number';
+  var labelString = '${labelPrepend} of interactions';
+  var chartX = ChartXAxe()
+    ..stacked = false
+    ..ticks = (LinearTickOptions()
+      ..minRotation = 0
+      ..maxRotation = 90);
+
+  var chartY = ChartYAxe()
+    ..stacked = false
+    ..scaleLabel = ScaleTitleOptions(labelString: labelString, display: true)
+    ..ticks = (LinearTickOptions()..min = 0);
+
+  var tooltipLabelCallback = (ChartTooltipItem tooltipItem, ChartData data) {
+    var xLabel = data.datasets[tooltipItem.datasetIndex].label;
+    var yLabel = tooltipItem.yLabel;
+    var suffix = dataNormalisationEnabled ? '%' : '';
+    return '${xLabel}: ${yLabel}${suffix}';
+  };
+
+  return ChartOptions(
+      responsive: true,
+      tooltips: ChartTooltipOptions(
+          mode: 'index',
+          callbacks:
+              ChartTooltipCallback(label: allowInterop(tooltipLabelCallback))),
+      legend: ChartLegendOptions(
+          position: 'bottom', labels: ChartLegendLabelOptions(boxWidth: 12)),
+      scales: ChartScales(display: true, xAxes: [chartX], yAxes: [chartY]));
+}
+
+ChartDataSets _generateTimeSeriesChartDataset(
+    String label, List<num> data, String lineColor) {
+  return ChartDataSets(
+      label: label,
+      fill: false,
+      backgroundColor: lineColor + 'aa',
```

Comment:
nit: consider using string interpolation:
```suggestion
      backgroundColor: '${lineColor}aa',
```

Reply:
Ah yes, fixed. Updated the bar chart color as well to use string interpolation.